tests/fixtures.py

```diff
@@ -41,17 +41,26 @@
         "standard_name": "time",
     },
 )
+# NOTE: With `decode_times=True`, the "calendar" and "units" attributes are
+# stored in `.encoding`.
+time_cf.encoding["calendar"] = "standard"
+time_cf.encoding["units"] = "days since 2000-01-01"
+
+
+# NOTE: With `decode_times=False`, the "calendar" and "units" attributes are
+# stored in `.attrs`.
 time_non_cf = xr.DataArray(
     data=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
     dims=["time"],
     attrs={
```

Comment:
Just bringing this to attention.

Reply:
I think this makes sense. This is a separate issue when I open a dataset with `xarray.open_mfdataset` and `decode_times=True`, I don't see anything populated in `ds.time.encoding` (but I do see calendar and units information stored here when using `xarray.open_dataset`).

Reply:
Nice catch! I found an open xarray issue related to `xr.open_mfdataset()` dropping encoding attributes.

https://github.com/pydata/xarray/issues/2436
https://github.com/pydata/xarray/issues/2436#issuecomment-449737841

\

Somebody mentions this about the issue here https://github.com/pydata/xarray/issues/2436#issuecomment-888204512:

> However, there is another problem related to xarray.open_mfdataset:
The encoding dictionary gets lost somewhere during the merging operation of the datasets of the respective files (https://github.com/pydata/xarray/issues/2436).
>
> This leads to problems for example with cf-xarray when trying to detect coordinates or bounds, but also leads to problems related to the time axis encoding apparently (as seen in the linked issue). I managed at least to avoid the problems for cf-xarray bounds and coordinates detection by using the decode functionality of xarray only after the datasets have been read in (leaving however the unnecessary time dimension in place ...):
> 
> ds = xarray.open_mfdataset("/path/to/files/*.nc")
> ds = xarray.decode_cf(ds, decode_coords="all")

---

**I think we might need to use the `preprocess` function to save the `.encoding` dict.**

Reply:
We can open up a separate issue for this. I'm glad you're finding all of these nonobvious behaviors/bugs in xarray and xCDAT.